### ISSUE TITLE
Correctly Authenticate with Strava API

### ISSUE THREAD

**Issue by josh-works** [josh-works/strava_run_polylines_osm#2]
This is hopelessly general, but here's where I'm at: 

![2021-06-25 at 9 29 AM](https://user-images.githubusercontent.com/5198260/123448329-09bc1b00-d598-11eb-86ab-b6a3a89cbc03.jpg)

When I run `> uvicorn authenticate:app --reload`, and visit `localhost:8000` I get:

![2021-06-25 at 9 41 AM](https://user-images.githubusercontent.com/5198260/123450098-93b8b380-d599-11eb-9a1f-331447444144.jpg)

------------------

Things I've tried/documented: https://github.com/josh-works/strava_run_polylines_osm#strava-api-basics

I just want to be able to do `python download_activities.py` and have it work.

Currently, I think `check_token()` is where this is breaking for me:

https://github.com/josh-works/strava_run_polylines_osm/blob/3f0fa6fb8c0266b4483118194b145ff0de010698/download_activities.py#L30-L31

https://github.com/josh-works/strava_run_polylines_osm/blob/3f0fa6fb8c0266b4483118194b145ff0de010698/download_activities.py#L11-L20

This blocks #1

**Comment by josh-works** [comment]
https://www.markhneedham.com/blog/2017/04/29/leaflet-strava-polylines-osm/

`Access token`: gives you access

`refresh token`: access has a lifetime, times out

I got access token here: https://github.com/josh-works/strava_run_polylines_osm#step-5-or-6-make-a-curl-request-to-exchange-the-authorization-code-and-scope-for-a-refresh-token-access-token-and-access-token-expiration-date-step-7a-from-the-graph-replace-the-client_secret-and-code-

Now store the tokens, so I can use the appropriate one at appropriate times.

**Comment by josh-works** [comment]
`Access token`: gives you access

`refresh token`: access has a lifetime, times out

I got access token here: https://github.com/josh-works/strava_run_polylines_osm#step-5-or-6-make-a-curl-request-to-exchange-the-authorization-code-and-scope-for-a-refresh-token-access-token-and-access-token-expiration-date-step-7a-from-the-graph-replace-the-client_secret-and-code-

**Comment by josh-works** [comment]
@mikegehard for the win!

TL;DR, this works, had to do a few things, like hardcode the `client_id` and `client_secret` in `download_activities.py`, and `token` in `extra_runs.py`. 

https://yizeng.me/2017/01/11/get-a-strava-api-access-token-with-write-permission/ was super helpful, too. Got all that working, brought it to Postman and got a working call:

![2021-06-25 at 1 05 PM](https://user-images.githubusercontent.com/5198260/123473631-0f285e00-d5b6-11eb-8cf8-78db40b7540c.jpg)

And with that access token in `extra_runs.py`, the API call worked and I got all the desired data!
